### ISSUE TITLE
Bug: `no-unreachable` reports export declarations that aren't evaluated where they appear

### ISSUE THREAD

**Issue by crimsonjay0** [eslint/eslint#21295]
### Environment

Node version: v24.20.0
npm version: v11.19.0
Local ESLint version: v10.9.1
Global ESLint version: v10.9.1
Operating System: Windows 11

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```js
export default [
  {
    rules: { "no-unreachable": "error" },
  },
];
```
</details>

```js
throw new Error();
export function foo() {}
export { foo as bar };
export * from "./other.js";
export var x;
```

### What did you expect to happen?

No errors. None of those four statements does any work at the position where it’s written. They’re hoisted or resolved during module instantiation, before the module body runs at all.

### What actually happened?

```
2:1  error  Unreachable code  no-unreachable
3:1  error  Unreachable code  no-unreachable
```

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoidGhyb3cgbmV3IEVycm9yKCk7XG5leHBvcnQgZnVuY3Rpb24gZm9vKCkge31cbmV4cG9ydCB7IGZvbyBhcyBiYXIgfTtcbmV4cG9ydCAqIGZyb20gXCIuL290aGVyLmpzXCI7XG5leHBvcnQgdmFyIHg7XG4iLCJvcHRpb25zIjp7InJ1bGVzIjp7Im5vLXVucmVhY2hhYmxlIjpbImVycm9yIl19LCJsYW5ndWFnZU9wdGlvbnMiOnsic291cmNlVHlwZSI6Im1vZHVsZSIsImVjbWFWZXJzaW9uIjoibGF0ZXN0IiwicGFyc2VyT3B0aW9ucyI6eyJlY21hRmVhdHVyZXMiOnt9fX19fQ==

### Participation

- [x] I am willing to submit a pull request for this issue.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by mdjermanovic** [comment]
I'm not sure it would be useful to not report these declarations. While they technically execute, the exports can never be accessed.

@eslint/eslint-team thoughts?

**Comment by snitin315** [comment]
I agree to keep the existing behavior. Closing this issue.
